examples/statistics/customized_violin_demo.py

```diff
@@ -18,87 +18,64 @@
 import numpy as np
 
 
-# functions to calculate percentiles and adjacent values
-def percentile(vals, p):
-    N = len(vals)
-    n = p*(N+1)
-    k = int(n)
-    d = n-k
-    if k <= 0:
-        return vals[0]
-    if k >= N:
-        return vals[N-1]
-    return vals[k-1] + d*(vals[k] - vals[k-1])
-
-
-def adjacent_values(vals):
-    q1 = percentile(vals, 0.25)
-    q3 = percentile(vals, 0.75)
-    iqr = q3 - q1  # inter-quartile range
-
-    # upper adjacent values
-    uav = q3 + iqr * 1.5
-    if uav > vals[-1]:
-        uav = vals[-1]
-    if uav < q3:
-        uav = q3
-
-    # lower adjacent values
-    lav = q1 - iqr * 1.5
-    if lav < vals[0]:
-        lav = vals[0]
-    if lav > q1:
-        lav = q1
-    return [lav, uav]
+def adjacent_values(vals, q1, q3):
+    upper_adjacent_value = q3 + (q3 - q1) * 1.5
+    upper_adjacent_value = np.clip(upper_adjacent_value, q3, vals[-1])
+
+    lower_adjacent_value = q1 - (q3 - q1) * 1.5
+    lower_adjacent_value = np.clip(lower_adjacent_value, vals[0], q1)
+    return [lower_adjacent_value, upper_adjacent_value]
```

Comment:
To avoid the unpacking, I could also optionally make 'whiskers' an np.array and then do:
`whiskersMin, whiskersMax = whiskers[:, 0], whiskers[:, 1]`